### PR TITLE
Counting label frequencies bug fix in PushRelabelMFImpl

### PR DESCRIPTION
**Pull request by d-michail**
This is a fix (and two tests cases) for issue #270. 

The problem was that the code which counted the label frequencies assumed that the input graph is strongly connected.

### CODE REVIEW

**Comment by jkinable** [comment]
@alexeykudinkin would you agree this fixes issue #270 ?

File: jgrapht-core/src/test/java/org/jgrapht/alg/flow/PushRelabelMinimumSTCutTest.java [hunk 1]
```diff
@@ -52,6 +55,46 @@
...
+    }
+    
+    public void testDisconnected()
+    {
+        SimpleDirectedWeightedGraph<Integer, DefaultWeightedEdge> network =
+            new SimpleDirectedWeightedGraph<>(DefaultWeightedEdge.class);
+        network.addVertex(0);
+        network.addVertex(1);
+        network.addVertex(2);
+        network.setEdgeWeight(network.addEdge(0, 1), 1.0);
```
Comment (jkinable) [inline jgrapht-core/src/test/java/org/jgrapht/alg/flow/PushRelabelMinimumSTCutTest.java]: perhaps a bit picky but I think it's cleaner to use the Graphs:
Graphs.addEdge(Graph<V,E> g, V sourceVertex, V targetVertex, double weight);
method.

**Comment by d-michail** [comment]
I cleaned it up a little. Now the capacities are all "one" as they do no matter on this one.

**Comment by alexeykudinkin** [comment]
Yeah, initialisation was obviously erroneous: it didn't consider the case of disjoint networks.
Awesome work @d-michail, thanks for tracking this down.

@jsichi i believe we can move forward with this, can't we?

**Comment by jkinable** [comment]
@alexeykudinkin yep, will merge it when I get home. Thank for verifying :)

**Comment by jkinable** [comment]
Thanks @d-michail  for fixing this!

**Pull request merged**